src/main/java/org/opennms/test/system/api/NewTestEnvironment.java

```diff
@@ -324,12 +333,42 @@ private void spawnPostgres() throws DockerException, InterruptedException, IOExc
             return;
         }
 
+        LOG.debug("Starting PostgreSQL");
+
         final Builder builder = HostConfig.builder()
                 .publishAllPorts(true);
         spawnContainer(alias, builder);
     }
 
     /**
+     * Spawns the Apache Kafka container.
+     */
+    private void spawnKafka() throws DockerException, InterruptedException, IOException {
+        final ContainerAlias alias = ContainerAlias.KAFKA;
+        if (!(isEnabled(alias) && isSpawned(alias))) {
+            return;
+        }
+
+        LOG.debug("Starting Kafka");
+
+        // Bind Kafka and Zookeeper to the same ports on the Docker host
+        final Map<String, List<PortBinding>> portBindings = new HashMap<String, List<PortBinding>>();
+        for (String port : new String[] { "2181", "9092" }) {
+            portBindings.put(port, Collections.singletonList(PortBinding.of("0.0.0.0", port)));
+        }
+
+        // Advertise Kafka on the Docker host address
+        List<String> env = Arrays.asList(new String[] {
+            "ADVERTISED_HOST=" + InetAddress.getLocalHost().getHostAddress(),
```

Comment:
by *should* I mean *could*, since I suspect there would be other changes necessary to relocate the containers to another host